crates/codegen/schema/src/validation/rules/references.rs

```diff
@@ -109,10 +116,14 @@ impl<'v> ReferencesCollector<'v> {
 
 #[derive(thiserror::Error, Debug)]
 enum Errors {
-    #[error("Production '{0}' is not defined.")]
-    NotFound(String),
-    #[error("Expression '{0}' cannot be used as a reference. References are only valid to top-level productions.")]
-    Illegal(String),
+    #[error("There is no Production '{0}'.")]
+    NoSuchProduction(String),
+    #[error("You cannot reference Node '{0}'.")]
+    NodesCannotBeReferenced(String),
+    #[error("Scanner references Parser '{0}'.")]
```

Comment:
Love this new rule!

Suggestion:

```suggestion
    #[error("Scanner cannot reference Parser '{0}'.")]
```

Reply:
done